xdcs-server/src/main/java/pl/edu/agh/xdcs/db/entity/WorkShape.java

```diff
@@ -0,0 +1,72 @@
+package pl.edu.agh.xdcs.db.entity;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import javax.persistence.AttributeConverter;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.List;
+
+@Getter
+@Builder
+@EqualsAndHashCode
+public class WorkShape {
+    private int x;
+    private int y;
+    private int z;
+
+    public static WorkShape fromList(List<Integer> shape) {
+        if (shape == null) {
```

Comment:
Ten if null return null do mnie nie przemawia, null powinien być sprawdzany najwcześniej jak to jest możliwe, a nie propagowany dalej

Reply:
zmienione